### PR TITLE
Rename classes - use NS instead of flat structure with "_"

### PR DESCRIPTION
**Pull request by georgehristov**
Split of https://github.com/atk4/dsql/pull/196

Part II - Pure refactoring of namespaces

### BC is provided for a short time, but not fully BC because `instanceof` operator like: `instanceof Query_MySQL` will fail with object instanced using the new naming

### CODE REVIEW

File: tests/OracleTest.php
Comment (mvorisek) [inline tests/OracleTest.php]: you use single quotes string here...

for prod/non test I qould use probably something like `['' => ::class, ...][$ver]`...

Comment (georgehristov) [reply tests/OracleTest.php]: What do you mean single quotes string?

**Comment by mvorisek** [comment]
@georgehristov rebase

image

why is there `c` suffix after the version?

File: tests/OracleTest.php
Comment (mvorisek) [reply tests/OracleTest.php]: `'\atk4\dsql\Oracle' . $version . '...` should be the preffered format

- we should use single quoted strings everywhere
- do not escape `\` when not needed

**Comment by georgehristov** [comment]
@mvorisek that is how it was referred to in the original class

**Comment by mvorisek** [comment]
@georgehristov seems like typo... Connection_Oracle12 is/was without it!

**Comment by georgehristov** [comment]
@mvorisek changed now

**Comment by georgehristov** [comment]
https://www.oracle.com/database/12c-database/

@mvorisek so it should be `Version12c` everywhere

File: tests/QueryTest.php
Comment (mvorisek) [inline tests/QueryTest.php]: `use atk4\dsql\Mysql` shoud be preffered instead. `as Query_MySQL` is too chatty without any value provided and hard to refactor

also fix RandomTest (and maybe other) to use ...\tests NS.

**Comment by mvorisek** [comment]
@georgehristov 
image

from Wiki, I would keep it without the suffix, just the version :)

Will be also consitent if we will need more DB specific classes for other engines.

**Comment by georgehristov** [comment]
@mvorisek changed now, officially they refer to it with the "c" for whatever reason so we follow official version name

**Comment by mvorisek** [comment]
No - see my update above, we want minimalistic name and consistency across classes

File: tests/QueryTest.php
Comment (georgehristov) [reply tests/QueryTest.php]: What do you mean with `use atk4\dsql\Mysql`?

Comment (mvorisek) [reply tests/QueryTest.php]: later refer to this import like `MySql\Connection` :) (never use aliases with _ if you can import the parent NS)

Comment (georgehristov) [reply tests/QueryTest.php]: Beautiful 😄

**Comment by mvorisek** [comment]
becase of https://github.com/atk4/ui/issues/1024 also rename the names to PascalCase:
- `MySQL` -> `Mysql`
- `SQLite` => `Sqlite`
etc. (the rule is - use letter capital when you will use it in snake case - ex. `mysql`, not `my_s_q_l`)

**Comment by georgehristov** [comment]
This we discussed already and we keep like it is for now

**Comment by mvorisek** [comment]
where - because it is planned like I propose :D

It would be better to do it now at once...

**Comment by georgehristov** [comment]
> where - because it is planned like I propose :D

where?

**Comment by mvorisek** [comment]
> we discussed already

**Comment by georgehristov** [comment]
I will not change it now as I do not like the Pascal case in this instance (DB names)
If agreed by all then we can make another PR.

**Comment by georgehristov** [comment]
> where - because it is planned like I propose :D

Where: Discord
When: 04/28/2020

**Comment by mvorisek** [comment]
image

I see approval to go ahead with "snake case reasoning" :)

so is this ok with you to do now so we can merge today?

**Comment by mvorisek** [comment]
- so ok with you to do it now?
- I also updated the desc - this PR is breaking and it may be very hard to discover it - ok to emit deprecated messages from the old classes when they are loaded?

**Comment by georgehristov** [comment]
Actually I don't like it but I would follow if there is enough support for it.
But having @Darkside666 say it is maybe okay is not sufficient to set the direction.
So this can be done at later stage or we discuss here before merging.

**Comment by mvorisek** [comment]
it is discussed - we should follow snake case reasoning, because not following it results in naming mess like `MySQLURL` instead of `MysqlUrl` (or mysql_url in snakecase, that is how would you normally also written it)

so this PR is complete except:
- [ ] naming normalization with snake_case rationale
- [x] emit deprecation warning in every deprecated class

File: src/Concerns/IsDeprecated.php
Comment (mvorisek) [inline src/Concerns/IsDeprecated.php]: why @ before, will it actually emit anything?

If constructor is not used, this will not trigger... I think it is better to move it before/after the class def directly and emit these warnings always when the class is inluded.

Comment (georgehristov) [reply src/Concerns/IsDeprecated.php]: The @ is added by CS Fixer.

I will move it in the file

File: src/Connection_Counter.php
Comment (mvorisek) [inline src/Connection_Counter.php]: You should always review CS fixes manually! I tested it now and with @ is does not emit any message!

If needed, write it as `'trigger_error'(...` (yes, with single quotes)

Comment (georgehristov) [reply src/Connection_Counter.php]: It does not work without @ either as warning is triggered immediately during testing caused by phpunit\php-code-coverage

Comment (mvorisek) [reply src/Connection_Counter.php]: https://3v4l.org/QepiX

Comment (georgehristov) [reply src/Connection_Counter.php]: This is clear. The problem is that the files are autoloaded by phpunit coverage so the warning is triggered during testing always

Comment (mvorisek) [reply src/Connection_Counter.php]: Does my "test" commit solves that?

**Comment by mvorisek** [comment]
> naming normalization with snake_case rationale

@georgehristov can you finish this based on the discussion above? Exlc. this I am for merging this immediatelly.

File: src/Postgre/Connection.php
Comment (mvorisek) [inline src/Postgre/Connection.php]: I just written my friend working with this database daily. He told me that only two names are used for this DB - Postgres and Postgresql. I also asked him about PgSQL -> Pgsql, but he told me, that they almost never use this name for it.

Postgres is an old/alias name for Postgresql based on Wiki. So I would you full "Postgresql" here. Wdyt?

Comment (georgehristov) [reply src/Postgre/Connection.php]: Aim is to be as intuitive and standard as possible and build the magic on top of that so `Postgresql` is fine with me.

**Review by mvorisek — APPROVED** [review]
Rename NS `Postgre` to `Postgresql` and merge. Good move for Atk, good work, @georgehristov!

(and update all dependent projects to use the new names)

File: src/Postgre/Connection.php
Comment (mvorisek) [inline src/Postgre/Connection.php]: filepath as well...

**Comment by georgehristov** [comment]
@mvorisek thanks for review and approval. I will not hurry to merge but await review/approval from @DarkSide666 as well

**Comment by DarkSide666** [comment]
Sorry guys, it's Midsummer holidays here :)
I will quickly review now.

About naming Mysql vs MySQL - as I said, I like MySQL better, but I see the point of @mvorisek. Probably for the future when hopefully we will support like 30-50 different DB engines it's better to set quite strict (even if not intuitive at first sight) naming principles.

**Review by DarkSide666 — APPROVED** [review]
OK I think it's fine.

**Comment by georgehristov** [comment]
@DarkSide666 at the moment in your code you can still use `MySQL\Connection` for instance (as I also prefer it being visually pleasing) as case is ignored.

**Comment by mvorisek** [comment]
No you can not! Because of CS autoloader on non Windows platforms!

**Pull request merged**
